### DISCUSSION TITLE
Release 1.2.0

### DISCUSSION THREAD

**Discussion by ThomasWaldmann** [borgbackup/borg#6337] [General]
## borgbackup 1.2.0 release

Long changelog (including upgrade and compatibility notes):

https://github.com/borgbackup/borg/blob/1.2.0/docs/changes.rst#version-120-2022-02-22-220222--

### Installation

If you use pip to install this, use: pip install pkgconfig ; pip install "borgbackup==1.2.0"

For other installation methods and more details, please see: https://borgbackup.org/

### Feedback

Discussion: ...

<hr /><em>This discussion was created from the release <a href='https://github.com/borgbackup/borg/releases/tag/1.2.0'>Release 1.2.0</a>.</em>

**Comment by goebbe** [comment]
Question: Are there any problems expected, when the borg clients are updated to 1.2 while the servers (NAS or mass storage providers) are still using borg 1.1?

**Reply by ThomasWaldmann** [reply]
No problems AFAIK.

But borg then will still have the old compaction behaviour (it will immediately compact implicitly at the end of each repo writing command).

**Reply by rustycanb** [reply]
This interesting because rsync.net is still on borg version 1.1.17 and will not upgrade until 1.2.1 is released. 

Running (1.2.0) `borg compact --cleanup-commits -v ssh://usernumber@mylogin.rsync.net/path/to/repo` results in even more 17b commit files. When available commands are listed at rsync.net, compact is not among them.

**Reply by ThomasWaldmann** [reply]
IIRC, you'll need >= 1.2 also on the server side (when running client-server) for `--cleanup-commits` or `compact` to work.

**Reply by rustycanb** [reply]
Thanks for your quick response. Not really a huge issue, just interesting. BTW, 1.2.0 is great, thanks...

**Comment by romario74** [comment]
The binary does not work on OL/CentOS/RHEL7?

`# ./borg-linux64 
[4496] Error loading Python lib '/tmp/_MEISskFjQ/libpython3.9.so.1.0': dlopen: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /tmp/_MEISskFjQ/libpython3.9.so.1.0)`

Anywhere I can get a working binary for it?

**Reply by infectormp** [reply]
https://github.com/borgbackup/borg/issues/6391

**Reply by romario74** [reply]
Thx, got it.

**Comment by knutov** [comment]
borg 1.2 prints nothing with `borg delete -P PREFIX --dry-run` or with `-a GLOB`. Is it known bug? In my case it actually deletes what it should by prefix or glob without `--dry-run`, but there is no way to see what will be deleted with `--dry-run`.

**Reply by ThomasWaldmann** [reply]
does it maybe need -v or --list?

**Reply by knutov** [reply]
yes, `--list` helps. Not obvious. Output is printed for `borg delete` without `--list`, why it is required only for `--dry-run` then?

**Reply by ThomasWaldmann** [reply]
file a bug if it behaves very different with/without dry-run, that is likely not intended.

**Reply by knutov** [reply]
looks like I had problem about options on left and right side with repo between.

`borg delete -P $PREFIX  -v $REPO` do not print anything without `--list`, so everething is fine.

**Comment by knutov** [comment]
Not sure, may be not 1.2 version specific, but `--glob-archive` does not working with regexp:

```
borg list user123.mysql --prefix user123-2 | awk '{print $1}'
user123-2022-03-20-065449
...
user123-2022-03-28-031228

borg list user123.mysql -a "re:^user123"
# empty output
```

In my scenario I have to select snapshots to prune with filter based on prefix (exact match) and suffix (regexp), Seems to be not possible now.

**Reply by ThomasWaldmann** [reply]
check the docs if that is supposed to work and if so, file a bug.

**Reply by knutov** [reply]
https://github.com/borgbackup/borg/issues/6504
